src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetQueryCriteriaTimePeriodsValidatorTests.cs

```diff
@@ -0,0 +1,338 @@
+            new() { Period = "2000/1999", Code = "AY" },
+        },
+        new DataSetQueryTimePeriod[]
+        {
+            new() { Period = "2022", Code = "" },
+            new() { Period = "2022/2023", Code = "" },
+            new() { Period = "2022", Code = "ay" },
+            new() { Period = "2022", Code = "Invalid" },
+            new() { Period = "2022", Code = "1" },
+            new() { Period = "2022", Code = "WEEK12" },
+            new() { Period = "2022/2023", Code = "JANUARY" },
+        },
+    };
+
+    public class EqTests : DataSetQueryCriteriaTimePeriodsValidatorTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidTimePeriodsSingle))]
+        public void Success(DataSetQueryTimePeriod timePeriod)
+        {
+            var query = new DataSetQueryCriteriaTimePeriods { Eq = timePeriod };
+
+            _validator.TestValidate(query).ShouldNotHaveAnyValidationErrors();
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidTimePeriodsSingle))]
+        public void Failure(DataSetQueryTimePeriod timePeriod)
+        {
+            var query = new DataSetQueryCriteriaTimePeriods { Eq = timePeriod };
+
+            var result = _validator.TestValidate(query);
+
+            Assert.All(result.Errors, error =>
+                Assert.StartsWith("Eq", error.PropertyName));
+        }
+    }
+
+    public class NotEqTests : DataSetQueryCriteriaTimePeriodsValidatorTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidTimePeriodsSingle))]
+        public void Success(DataSetQueryTimePeriod timePeriod)
+        {
+            var query = new DataSetQueryCriteriaTimePeriods { NotEq = timePeriod };
+
+            _validator.TestValidate(query).ShouldNotHaveAnyValidationErrors();
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidTimePeriodsSingle))]
+        public void Failure(DataSetQueryTimePeriod timePeriod)
+        {
+            var query = new DataSetQueryCriteriaTimePeriods { NotEq = timePeriod };
+
+            var result = _validator.TestValidate(query);
+
+            Assert.All(result.Errors, error =>
+                Assert.StartsWith("NotEq", error.PropertyName));
+        }
+    }
+
+    public class InTests : DataSetQueryCriteriaTimePeriodsValidatorTests
+    {
+        [Theory]
+        [MemberData(nameof(ValidTimePeriodsMultiple))]
+        public void Success(DataSetQueryTimePeriod[] timePeriods)
+        {
+            var testObj = new DataSetQueryCriteriaTimePeriods { In = timePeriods };
+
+            _validator.TestValidate(testObj).ShouldNotHaveAnyValidationErrors();
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidTimePeriodsMultiple))]
+        public void Failure(DataSetQueryTimePeriod[] timePeriods)
+        {
+            var query = new DataSetQueryCriteriaTimePeriods { In = timePeriods };
+
+            var result = _validator.TestValidate(query);
+
+            Assert.All(result.Errors, error =>
```

Comment:
Might be worth counting the number of errors and checking it's equal to the number of items in each list in `InvalidTimePeriodsMultiple`

Same for other tests in this class and the others where we are testing for multiple errors